extensions/amp-form/0.1/amp-form.js

```diff
@@ -556,14 +556,14 @@ export class AmpForm {
       // Validity checking should always occur, novalidate only circumvent
       // reporting and blocking submission on non-valid forms.
       const isValid = checkUserValidityOnSubmission(this.form_);
-      if (this.shouldValidate_ && !isValid) {
+      if (this.shouldValidate_) {
```

Comment:
Right, sorry for not clarifying. All of the validators will hide displayed validations on input or blur, so by the time `report()` is called due to form submission it'll be a no-op.

I think this conditional existed because abstractly there's no point to invoking a validator on a form you already know is valid (as opposed to due to avoiding some change in behavior).